src/lib.rs

```diff
@@ -3,6 +3,7 @@
 #![deny(unsafe_code)]
 // TODO: Change this to enable deny(unused) if IPv6 or IPv4 are enabled
 #![cfg_attr(feature = "proto-ipv4", deny(unused))]
+#![feature(slice_patterns)]
```

Comment:
`slice_patterns` isn't available on stable, so if we're going to use slice patterns an additional note should probably be added to https://github.com/m-labs/smoltcp/blob/master/.travis.yml#L4-L9

Reply:
added